tests/ftests/071-sudo-set_default_systemd_cgroup.py

```diff
@@ -42,131 +36,144 @@ def prereqs(config):
+        cause = '\n'.join(filter(None, [cause, tmp_cause]))
 
-    # Emulate the systemd slice/scope creation
-    f = open(SYSTEMD_DEFAULT_CGROUP_FILE, 'w')
-    f.write(os.path.join(SLICE, SCOPE))
-    f.close()
+    cli_pid = CgroupCli.get_pids_in_cgroup(config, OTHER_CGNAME, CONTROLLER)[0]
 
-    if not os.path.exists(SYSTEMD_DEFAULT_CGROUP_FILE):
+    if pid != cli_pid:
         result = consts.TEST_FAILED
-        cause = 'Failed to create %s' % SYSTEMD_DEFAULT_CGROUP_FILE
-        return result, cause
-
-    # create /sys/fs/cgroup/cpu/libcgtests.slice (v1) or /sys/fs/cgroup/libcgtests.slice (v2)
-    result, cause = create_cgrp(config, SLICE, ignore_systemd=True)
-    if result == consts.TEST_FAILED:
-        return result, cause
-
-    # create /sys/fs/cgroup/cpu/libcgtests.slice/test071.scope (v1) or
-    #        /sys/fs/cgroup/libcgtests.slice/tests071.scope (v2)
-    result, cause = create_cgrp(config, os.path.join(SLICE, SCOPE), ignore_systemd=True)
-    if result == consts.TEST_FAILED:
-        return result, cause
-
-    # In hybrid mode /sys/fs/cgroup/unified/libcgtests.slice/test071.scope
-    # is created by the systemd and we relay on it for checking if the
-    # slice and scope were set as default systemd cgroup (new cgroup root)
-    if MODE == Mode.CGROUP_MODE_HYBRID:
-        result, cause = create_cgrp(config, SLICE, None, ignore_systemd=True)
-        if result == consts.TEST_FAILED:
-            return result, cause
-
-        result, cause = create_cgrp(config, os.path.join(SLICE, SCOPE), None, ignore_systemd=True)
-
-    return result, cause
+        tmp_cause = 'Expected pid {} to be in {} cgroup, but received pid {} ' \
+                    'via CLI instead'.format(pid, OTHER_CGNAME, cli_pid)
+        cause = '\n'.join(filter(None, [cause, tmp_cause]))
+
+    Process.kill(config, pid)
+    cg.delete()
+
+    #
+    # Test 3 - Write the slice/scope and attempt to set them as the default.
+    #          This should fail because they haven't been created yet, and thus
+    #          it's an invalid path
+    #
+    try:
+        Cgroup.write_default_systemd_scope(SLICE, SCOPE, True)
+    except RuntimeError as re:
+        if 'Failed to set' not in str(re):
+            result = consts.TEST_FAILED
+            tmp_cause = 'Expected \'Failed to set\' to be in the exception, ' \
+                        'received {}'.format(str(re))
+            cause = '\n'.join(filter(None, [cause, tmp_cause]))
+    else:
+        result = consts.TEST_FAILED
+        tmp_cause = 'write_default_systemd_scope() erroneously passed'
+        cause = '\n'.join(filter(None, [cause, tmp_cause]))
 
+    #
+    # Test 4 - Create a systemd scope and set it as the default.  Everything
+    #          should work properly in this case
+    #
+    pid = None
+    if Cgroup.cgroup_mode() != Mode.CGROUP_MODE_LEGACY:
+        pid = config.process.create_process(config)
+        Cgroup.create_scope(SCOPE, SLICE, pid=pid)
```

Comment:
The initial idea was to emulate `create_scope()`,  so that users can use one of their hand-created `slice` and `scope`

Reply:
I see.  Do you want me to rework this to match your original goal?  Do these changes sufficiently test the code?

Reply:
The changes will test the `default systemd slice` so I don't have a strong opinion. We should be good, with `create_scope()`